### PR TITLE
Fix select label override

### PR DESCRIPTION
**Pull request by benCoomes**
Fixes bug introduced in #12 where label matching override checks for CI and reviews.

Also:
* checks labels first so that if labels don't match, API calls for PR status can be skipped.
* Creates test helper functions to reduce repeated object creation

### CODE REVIEW

**Comment by benCoomes** [comment]
I was scratching my head for a while because I couldn't figure out why tests were failing after reordering the CI/label checks - turns out it was because when the CI checks were skipped for one PR, the graphql mocks got out of order. 

Is there a way to make mocks return a value based on a parameter, rather than setting them up to be returned values in a sequence?

**Comment by GrantBirki** [comment]
@benCoomes When this PR merges, will it fully resolve https://github.com/github/combine-prs/issues/11?

**Comment by benCoomes** [comment]
> This is probably a side effect of two things: 1. Me using jest 2. Me not really understanding jest.

Ha, we are in the same boat then! I'll spend some time to see what I can do.

> When this PR merges, will it fully resolve https://github.com/github/combine-prs/issues/11?

It will give a workaround for it, so I'd be okay to close the issue. Dependabot needs to be configured per branch anyways, so setting a branch-specific label and then having a `combine-pr` config to select on that branch label is not hard to set up. 

It'd still be nice to have it separate PRs by target branch by default, but I think the config would need to change to support either:
* creation of multiple PRs & branches in a single run
* or, limit each run to a single target branch

File: __tests__/main.test.js [hunk 1]
```diff
@@ -134,108 +134,24 @@ beforeEach(() => {
...
-                    }
-                  ]
-                }
-              }
-            }
-          }
-        }),
+      graphql: jest.fn().mockImplementation((_query, params) => {
+        switch (params.pull_number) {
+          case 1:
```
Comment (GrantBirki) [inline __tests__/main.test.js:61]: I love this! TIL

**Review by GrantBirki — APPROVED** [review]
Amazing! 🎉 Thank you @benCoomes, you have significantly improved the test suite here as well and I learned a few new things following through your commits as well 🙇

**Pull request merged**